experiment-setup/bootstrap/provisioning/server-provisioning.sh

```diff
@@ -0,0 +1,41 @@
+#!/bin/sh
+check_command() {
+    if [ $? -ne 0 ]; then
+        echo "Error: $1 failed."
+        exit 1
+    fi
+}
+
+echo "Starting provisioning for Docker on Alpine Linux..."
+
+apk update
+check_command "apk update"
+
+echo "Installing Docker and openrc (manager)..."
+apk add --no-cache openrc docker
+check_command "Docker installation"
+
+echo "Installing Docker Compose..."
+apk add --no-cache docker-compose
+check_command "Docker Compose installation"
+
+echo "Enabling and starting Docker service ..."
+rc-update add docker default # or boot here...
+check_command "Install and set service"
+
+echo "Provisioning complete. Docker and Docker Compose are installed."
+
+echo "Starting Docker..."
+service docker start # if this guy don't work then we have a problem
+
+echo "Pulling the images!"
+
+DOCKER_IMAGES="dudustri/waect-rust-actix alpine:edge" #list the images here
+
+for IMAGE in $DOCKER_IMAGES; do
```

Comment:
I don't get why we loop over different docker images, can you explain? In my head we just pull the latest image per application and then execute the client.

Reply:
We are pulling all the images at once before start the experiment. So we gonna have all the images in memory to spin up quickly.